docs/guide/install-and-run.asciidoc

```diff
@@ -196,3 +201,263 @@ More information is available in {apm-server-ref-v}/configuring-howto-apm-server
 Don't forget to also read about
 {apm-server-ref-v}/securing-apm-server.html[securing APM Server], and
 {apm-server-ref-v}/monitoring.html[monitoring APM Server].
+
+[[quick-start-overview]]
+=== Quick start development environment
+
+ifeval::["{release-state}"=="unreleased"]
+
+Version {version} of APM Server has not yet been released.
+
+endif::[]
+
+ifeval::["{release-state}"!="unreleased"]
+
+If you're just looking for a quick way to try out Elastic APM, you can easily get started with Docker.
+Just follow the steps below.
+
+**Create a docker-compose.yml file**
+
+The https://www.docker.elastic.co/[Elastic Docker registry] contains Docker images for all of the products
+in the {stack}.
+You can use Docker compose to easily get the default distributions of {es}, {kib},
+and APM Server up and running in Docker.
+
+Create a `docker-compose.yml` file and copy and paste in the following:
+
+["source","yaml",subs="attributes"]
+--------------------------------------------
+{
```

Comment:
A few thoughts:

 - If we're going to put this front and centre in the docs, I think we should reformat it as "native" YAML, which most people would expect, rather than JSON-which-happens-to-be-YAML
 - There's a bunch of command args passed to apm-server that are not relevant to a "quick start" experience, like enabling profiling, rate-limiting, self-instrumentation. Having all of those makes it more difficult for people to understand, which invariably leads to cargo-culting
 - I'd be inclined to have the docker-compose.yml file checked in (and versioned), so it can be tested in CI

Reply:
Thanks Andrew!

> There's a bunch of command args passed to apm-server that are not relevant to a "quick start" experience, like enabling profiling, rate-limiting, self-instrumentation.

Makes sense. I've removed those below, but I'd need additional input on what else, if anything, you felt didn't belong.
<details>
  <summary>YAML</summary>

```yml
networks:
  default:
    name: elastic-apm
services:
  apm-server:
    cap_add:
    - CHOWN
    - DAC_OVERRIDE
    - SETGID
    - SETUID
    cap_drop:
    - ALL
    command:
    - apm-server
    - "-e"
    - "-E"
    - apm-server.rum.enabled=true
    - "-E"
    - apm-server.rum.event_rate.limit=1000
    - "-E"
    - apm-server.host=0.0.0.0:8200
    - "-E"
    - apm-server.read_timeout=1m
    - "-E"
    - apm-server.shutdown_timeout=2m
    - "-E"
    - apm-server.write_timeout=1m
    - "-E"
    - logging.json=true
    - "-E"
    - logging.metrics.enabled=false
    - "-E"
    - setup.kibana.host=kibana:5601
    - "-E"
    - setup.template.settings.index.number_of_replicas=0
    - "-E"
    - setup.template.settings.index.number_of_shards=1
    - "-E"
    - setup.template.settings.index.refresh_interval=1ms
    - "-E"
    - apm-server.kibana.enabled=true
    - "-E"
    - apm-server.kibana.host=kibana:5601
    - "-E"
    - apm-server.agent.config.cache.expiration=30s
    - "-E"
    - output.elasticsearch.hosts=["elasticsearch:9200"]
    - "-E"
    - output.elasticsearch.enabled=true
    - "-E"
    - 'output.elasticsearch.pipelines=[{pipeline: ''apm''}]'
    - "-E"
    - apm-server.register.ingest.pipeline.enabled=true
    container_name: localtesting_${STACK_VERSION:-7.5.1}_apm-server
    depends_on:
      elasticsearch:
        condition: service_healthy
      kibana:
        condition: service_healthy
    healthcheck:
      interval: 10s
      retries: 12
      test:
      - CMD
      - curl
      - "--write-out"
      - "'HTTP %{http_code}'"
      - "--fail"
      - "--silent"
      - "--output"
      - "/dev/null"
      - http://localhost:8200/
    image: docker.elastic.co/apm/apm-server:${STACK_VERSION:-7.5.1}
    labels:
    - co.elastic.apm.stack-version=${STACK_VERSION:-7.5.1}
    logging:
      driver: json-file
      options:
        max-file: '5'
        max-size: 2m
    ports:
    - 127.0.0.1:8200:8200
    - 127.0.0.1:6060:6060
  elasticsearch:
    container_name: localtesting_${STACK_VERSION:-7.5.1}_elasticsearch
    environment:
    - bootstrap.memory_lock=true
    - cluster.name=docker-cluster
    - cluster.routing.allocation.disk.threshold_enabled=false
    - discovery.type=single-node
    - path.repo=/usr/share/elasticsearch/data/backups
    - ES_JAVA_OPTS=-XX:UseAVX=2 -Xms1g -Xmx1g
    - path.data=/usr/share/elasticsearch/data/${STACK_VERSION:-7.5.1}
    - xpack.security.enabled=false
    - xpack.license.self_generated.type=trial
    - xpack.monitoring.collection.enabled=true
    healthcheck:
      interval: '20'
      retries: 10
      test:
      - CMD-SHELL
      - curl -s http://localhost:9200/_cluster/health | grep -vq '"status":"red"'
    image: docker.elastic.co/elasticsearch/elasticsearch:${STACK_VERSION:-7.5.1}
    labels:
    - co.elastic.apm.stack-version=${STACK_VERSION:-7.5.1}
    logging:
      driver: json-file
      options:
        max-file: '5'
        max-size: 2m
    ports:
    - 127.0.0.1:9200:9200
    ulimits:
      memlock:
        hard: -1
        soft: -1
    volumes:
    - esdata:/usr/share/elasticsearch/data
  kibana:
    container_name: localtesting_${STACK_VERSION:-7.5.1}_kibana
    depends_on:
      elasticsearch:
        condition: service_healthy
    environment:
      ELASTICSEARCH_URL: elasticsearch:9200
      SERVER_NAME: kibana.example.org
    healthcheck:
      interval: 10s
      retries: 20
      test:
      - CMD
      - curl
      - "--write-out"
      - "'HTTP %{http_code}'"
      - "--fail"
      - "--silent"
      - "--output"
      - "/dev/null"
      - http://kibana:5601/api/status
    image: docker.elastic.co/kibana/kibana:${STACK_VERSION:-7.5.1}
    labels:
    - co.elastic.apm.stack-version=${STACK_VERSION:-7.5.1}
    logging:
      driver: json-file
      options:
        max-file: '5'
        max-size: 2m
    ports:
    - 127.0.0.1:5601:5601
version: '2.1'
volumes:
  esdata:
    driver: local
  pgdata:
    driver: local
```

</details>

> I'd be inclined to have the docker-compose.yml file checked in (and versioned), so it can be tested in CI

This would be great. How do I make that happen?

Reply:
Here's what I came up with: https://gist.github.com/axw/05e6fb7f97108cb752b3051a0b304d2b

> > I'd be inclined to have the docker-compose.yml file checked in (and versioned), so it can be tested in CI
>
> This would be great. How do I make that happen?

That's not on you, I just think it's the way we should go :)

The steps I think we'd need to take are:
 1. Modify apm-integration-testing's compose script to:
    - Generate more human-readable YAML
    - Stop setting redundant config (there are a bunch of defaults which are being explicitly specified)
    - Provide a means of setting only the minimally required config
 2. As part of the release process:
    - Run the compose script
    - Smoke test the resulting docker-compose.yml
    - Check the file into elastic/apm-server

I'd personally find (1) helpful anyway, as I sometimes want to modify the docker-compose.yml by hand and find it hard to navigate.

@graphaelli do you think it's worthwhile?

Reply:
I agree the yaml is so much nicer to read and edit by hand.  I also like the idea of revisiting the redundant output - we broke the ability to launch older (some 6.x) clusters so we might as well clean things up too.  I'm all for this assuming it's not a huge effort, as long as yaml is optional - the main win right now with json is that it's built into python's stdlib and yaml support is not.

Reply:
I've created https://github.com/elastic/apm-integration-testing/issues/712 to track that.

Reply:
Thanks for the work on this Andrew. I've created a new `docker-compose.yml` file with your recommended yaml. I chatted with @graphaelli today 🏀, and he recommended we move forward with this PR for now, and update when elastic/apm-integration-testing#712 is ready.

I've also updated `RELEASES.md` with a reminder for myself to smoke test the yaml file before each major release.

Reply:
Thanks, sounds good!